test/suite/test_chunkcache04.py

```diff
@@ -0,0 +1,139 @@
+- Verify that when ingesting new chunks with old pinned objects, we are releasing the pin on the old objects. 
+'''
+class test_chunkcache4(wttest.WiredTigerTestCase):
+    rows = 10000
+
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('row_string', dict(key_format='S', value_format='S')),
+    ]
+
+    cache_types = [('in-memory', dict(chunk_cache_type='DRAM'))]
+    if sys.byteorder == 'little':
+        # WT's filesystem layer doesn't support mmap on big-endian platforms.
+        cache_types.append(('on-disk', dict(chunk_cache_type='FILE')))
+
+    scenarios = make_scenarios(format_values, cache_types)
+
+    def conn_config(self):
+        if not os.path.exists('bucket2'):
+            os.mkdir('bucket2')
+
+        return 'tiered_storage=(auth_token=Secret,bucket=bucket2,bucket_prefix=pfx_,name=dir_store),' \
+            'chunk_cache=[enabled=true,chunk_size=512KB,capacity=20MB,pinned=("table:chunkcache01", "table:chunkcache02"),type=DRAM,storage_path=WiredTigerChunkCache]'
+
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        extlist.extension('storage_sources', 'dir_store')
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def insert(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor[ds.key(i)] = str(i) * 100
+    
+    def read_and_verify(self, uri, ds):
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, self.rows):
+            cursor.set_key(ds.key(i))
+            cursor.search()
+            self.assertEqual(cursor.get_value(), str(i) * 100)
+
+    def test_chunkcache04(self):
+        uris = ["table:chunkcache03", "table:chunkcache04"]
+        ds = [SimpleDataSet(self, uri, 0, key_format=self.key_format, value_format=self.value_format) for uri in uris]
+
+        # Insert unpinned data into two tables. 
+        for i, dataset in enumerate(ds):
+            dataset.populate()
+            self.insert(uris[i], dataset)
+
+        # As we have not flushed yet, assert we have no newly inserted chunks. 
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted), 0)
+        
+        self.session.checkpoint()
+        self.session.checkpoint('flush_tier=(enabled)')
+
+        # Assert that chunks are not pinned. 
+        self.assertEqual(self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned), 0)
+
+        # Assert the new chunks are ingested. 
+        unpinned_ingest = self.get_stat(wiredtiger.stat.conn.chunk_cache_newly_inserted)
+        self.assertGreater(unpinned_ingest, 0)
+        
+        uris = ["table:chunkcache01", "table:chunkcache02"]
```

Comment:
These would be better in `self.pinned_uris`, that way you can reuse them here and in `conn_config`.

Reply:
your wish is my command [d0f1554](https://github.com/wiredtiger/wiredtiger/pull/9752/commits/d0f1554a1a4b797a5461c495bbecad346372b776)